### PR TITLE
Openshift4 Migration and CICD setup.

### PR DESCRIPTION
**Pull request by pwei1018**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

File: colin-api/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+ENV VCS_REF=${VCS_REF}
+ENV BUILD_DATE=${BUILD_DATE}
+
+LABEL org.label-schema.vcs-ref=${VCS_REF} \
+    org.label-schema.build-date=${BUILD_DATE}
+
+USER root
+
+# Create working directory
```
Comment (thorwolpert) [inline colin-api/Dockerfile:14]: colin will need the Oracle client files

File: colin-api/openshift/templates/bc.yaml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        FROM docker-remote.artifacts.developer.gov.bc.ca/python:3.8.6-buster
+
+        ARG BUILD_DATE
+        ARG VCS_REF
+
+        LABEL build_date=$BUILD_DATE
+        LABEL vcs_ref=$VCS_REF
+
+        USER root
+
```
Comment (thorwolpert) [inline colin-api/openshift/templates/bc.yaml:48]: this will need the oracle client libs

File: colin-api/openshift/templates/dc.yaml [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    value: dev
+    required: true
+
+  - name: ROLE
+    displayName: Role
+    description: Role
+    required: true
+    value: api
+
+  - name: NAMESPACE
```
Comment (thorwolpert) [inline colin-api/openshift/templates/dc.yaml:144]: this will need the oracle envvars

File: colin-api/requirements.txt [hunk 4]
```diff
@@ -6,6 +6,7 @@ Flask-SQLAlchemy
 Flask-Marshmallow
 Flask-OIDC
 flask-jwt-oidc
+Werkzeug==0.16.0
```
Comment (thorwolpert) [inline colin-api/requirements.txt:9]: can remove this, should be the newer one now

**Review by thorwolpert — COMMENTED** [review]
couple of comments

File: colin-api/requirements.txt (hunk 4 above)
Comment (thorwolpert) [reply colin-api/requirements.txt:9]: we'll open a techdeb tix for these

**Review by thorwolpert — APPROVED** [review]

File: colin-api/requirements.txt (hunk 4 above)
Comment (pwei1018) [reply colin-api/requirements.txt:9]: I tried other version of Wekzeug but all get error. And also tried upgrade the flask-jwt-oidc to newest version. Same, not sure which library had issue.

File: colin-api/openshift/templates/dc.yaml (hunk 3 above)
Comment (pwei1018) [reply colin-api/openshift/templates/dc.yaml:144]: envvars will manage by 1password. So, I removed all envvars from dc template.

**Pull request merged**